sycl/test/basic_tests/select_device.cpp

```diff
@@ -0,0 +1,71 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %t.out
+// RUN: env SYCL_DEVICE_TRIPLE=cpu %t.out
+// RUN: env SYCL_DEVICE_TRIPLE=gpu:level0 %t.out
+// RUN: env SYCL_DEVICE_TRIPLE=gpu:opencl %t.out
+// RUN: env SYCL_DEVICE_TRIPLE=cpu,gpu:level0 %t.out
+//
+// Checks if only specified device types can be acquired from select_device
+// when SYCL_DEVICE_TRIPLE is set
+// Checks that no device is selected when no device of desired type is
+// available.
+// UNSUPPORTED: windows
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+using namespace cl::sycl;
+
+int main() {
+  const char *envVal = std::getenv("SYCL_DEVICE_TRIPLE");
+  std::string forcedPIs;
+  if (envVal) {
+    std::cout << "SYCL_DEVICE_TRIPLE=" << envVal << std::endl;
+    forcedPIs = envVal;
+  }
+  if (!envVal || forcedPIs.find("gpu:level0") != std::string::npos) {
+    default_selector ds;
+    device d = ds.select_device();
+    std::cout << "Level-zero GPU Device is found: " << std::boolalpha
+              << d.is_gpu() << std::endl;
```

Comment:
> // Checks if only specified device types can be acquired from select_device
> // when SYCL_DEVICE_TRIPLE is set

a check that gpu opencl is not found is missing

Reply:
I see conversation marked "resolved", but I think the test case is weak. d.is_gpu() can return true even if no level-zero device is available, but only OpenCL is, and the test would pass.
This may be considered a NIT, but it would be really great to make the test stronger if possible.